src/Microsoft.AspNet.Mvc.Razor/Compilation/CompilationFailedException.cs

```diff
@@ -19,12 +20,19 @@ public class CompilationFailedException : Exception, ICompilationException
         /// details of the compilation failure.</param>
         public CompilationFailedException(
                 [NotNull] ICompilationFailure compilationFailure)
-            : base(Resources.FormatCompilationFailed(compilationFailure.SourceFilePath))
+            : base(FormatMessage(compilationFailure))
         {
             CompilationFailures = new[] { compilationFailure };
         }
 
         /// <inheritdoc />
         public IEnumerable<ICompilationFailure> CompilationFailures { get; }
+
+        private static string FormatMessage(ICompilationFailure compilationFailure)
+        {
+            return Resources.FormatCompilationFailed(compilationFailure.SourceFilePath) +
+                   Environment.NewLine +
+                   string.Join(Environment.NewLine, compilationFailure.Messages.Select(message => message.Message));
```

Comment:
Updated the FormattedMessage fror `RazorCompilationMessage` - so `exception.Message` now looks like:

```C#
One or more compilation failures occured:
/Views/Shared/MyView.cshtml (434,0) Expected "}".
```